test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+import com.sun.source.util.JavacTask;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol;
+import jdk.internal.shellsupport.doc.JavadocHelper;
+
+import javax.lang.model.element.*;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.*;
+import java.io.IOException;
+import java.lang.Runtime.Version;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SinceCheckerHelper {
+
+    //these are methods that were preview in JDK 13 and JDK 14, before the introduction
+    //of the @PreviewFeature
+    static final Set<String> LEGACY_PREVIEW_METHODS = Set.of(
+            "method:java.lang.String:stripIndent:()",
+            "method:java.lang.String:translateEscapes:()",
+            "method:java.lang.String:formatted:(java.lang.Object[])");
+
+    static final int JDK9 = 9, JDK23 = 23;
+    static final String JDK13 = "13",JDK14="14";
+
+    // only one hashmap is enough for now
+    public static Map<String, IntroducedIn> classDictionary = new HashMap<>();
```

Comment:
Meta-comment - limit access to everything in this class as much as possible. It is not really critical inside of tests, but it is a good style to use `private` if possible, etc. Also, drop the `static` on the map.